template/go/Dockerfile

```diff
@@ -1,5 +1,5 @@
 FROM --platform=${TARGETPLATFORM:-linux/amd64} ghcr.io/openfaas/classic-watchdog:0.1.5 as watchdog
-FROM --platform=${BUILDPLATFORM:-linux/amd64} golang:1.15-alpine3.13 as builder
+FROM --platform=${BUILDPLATFORM:-linux/amd64} golang:1.17-alpine3.13 as builder
```

Comment:
We would be able to sign off on Go 1.16, but 1.17 is untested.

A PR would be needed for this repo also https://github.com/openfaas/golang-http-template

Reply:
Perfect, That's the template I have been working on for a PoC serverless API